go/interpreter/lfvm/instructions.go

```diff
@@ -924,19 +903,30 @@ func genericCreate(c *context, kind tosca.CallKind) {
 		return
 	}
 
-	if c.memory.expandMemory(offset.Uint64(), size.Uint64(), c) != nil {
+	sizeU64 := size.Uint64()
+	if c.memory.expandMemory(offset.Uint64(), sizeU64, c) != nil {
 		c.signalError()
 		return
 	}
 
-	if err := checkInitCodeSize(c, size); err != nil {
-		c.signalError()
-		return
+	if c.isAtLeast(tosca.R12_Shanghai) {
+		if checkMaxInitCodeSize(sizeU64) != nil {
+			c.useGas(c.gas)
```

Comment:
Use gas is not a responsibility here. 
Any out of gas error must lead to all gas being consumed, and deal with in a single point of responsibility manner. 
This is line is redundant, having it may hide severe errors somewhere else.